pydantic/types.py

```diff
@@ -122,15 +122,18 @@ class ConstrainedList(list):  # type: ignore
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
-        yield list_validator
         yield cls.list_length_validator
 
     @classmethod
     def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
         update_not_none(field_schema, minItems=cls.min_items, maxItems=cls.max_items)
 
     @classmethod
-    def list_length_validator(cls, v: 'List[T]') -> 'List[T]':
+    def list_length_validator(cls, v: 'Optional[List[T]]', field: 'ModelField') -> 'Optional[List[T]]':  # type: ignore[name-defined] # noqa: E501,F821
```

Comment:
@samuelcolvin Thank you for the review ! Just a question on the mypy policy as I could
- not set the type of `field` and just add `# type: ignore`
- same but explaining the ignored error but then add a `noqa` for length
- do some refactoring to avoid circular imports (overkill probably)
- let it like that

Reply:
you just needed to import `ModelField`, imports inside `if TYPE_CHECKING:` don't cause circular import problems.